src/main/scala/com/recipepuppy/query.scala

```diff
@@ -0,0 +1,28 @@
+package com
+package recipepuppy
+
+import cats.implicits._
+import cats.effect._
+import org.http4s.Uri
+
+import pesto.http.fetch
+
+object query {
+  val baseUrl = Uri.uri("http://www.recipepuppy.com/api/")
+
+  def apply(ingredient: String, dish: String, page: Int = 0): IO[Vector[Recipe]] = {
+    val uri = baseUrl +? ("i", ingredient) +? ("q", dish) +? ("p", page)
+
+    fetch[ResultPage](uri)
+      .flatMap(
+        rp =>
+          if (rp.results.isEmpty) {
+            IO.pure(rp.results)
```

Comment:
`rp.results.pure[IO]` is another option here. This generalizes to any Applicative